deeplake/util/transform.py

```diff
@@ -532,3 +535,46 @@ def process_transform_result(result: List[Dict]):
         for key in keys:
             final[key].append(item[key])
     return final
+
+
+def rechunk_if_necessary(ds):
+    with ds:
+        for tensor in ds.tensors:
+            try:
+                tensor = ds[tensor]
+            # temp tensors
+            except TensorDoesNotExistError:
+                continue
+            if not tensor.meta.sample_compression and not tensor.meta.chunk_compression:
+                engine = tensor.chunk_engine
+                num_chunks = engine.num_chunks
+                if num_chunks > 1:
+                    num_samples = engine.num_samples
+                    max_shape = tensor.meta.max_shape
+                    if len(max_shape) > 0:
+                        nbytes = (
```

Comment:
these 2 lines can be refactored out into a method in chunk_engine called get_avg_chunk_size